src/WebJobs.Script/Description/Workers/WorkerFunctionDescriptorProvider.cs

```diff
@@ -129,11 +129,16 @@ internal bool BindingAttributeContainsExpression(IEnumerable<FunctionBinding> bi
                 {
                     return attribute.GetType()
                                     .GetProperties(BindingFlags.Instance | BindingFlags.Public)
-                                    .Any(prop => prop.PropertyType == typeof(string) && _expressionRegex.IsMatch((string)prop.GetValue(attribute)));
+                                    .Any(prop => prop.PropertyType == typeof(string) && IsMatch((string)prop.GetValue(attribute)));
                 }
             }
 
             return false;
         }
+
+        private bool IsMatch(string value)
+        {
+            return string.IsNullOrEmpty(value) ? false : _expressionRegex.IsMatch(value);
```

Comment:
The specific scenario for the bug is - if there is no expression in `BlobPath` and the connection is null then the regex.IsMatch(connection) will throw null exception. If there is an expression in `BlobPath` then it gets returned as true and the empty connection is not evaluated so we won't see the error.